### PR TITLE
Oh 685

### PR DESCRIPTION
**Pull request by mikaelbarlund**

### CODE REVIEW

File: organisaatio-service/src/main/java/fi/vm/sade/organisaatio/resource/impl/v4/OrganisaatioResourceImplV4.java [hunk 1]
```diff
@@ -208,6 +217,54 @@ public OrganisaatioHakutulosV4 findDescendants(String oid) {
...
+        } catch (NotAuthorizedException nae) {
+            LOG.warn("Not authorized to read organisation: " + oid);
+            throw new OrganisaatioResourceException(HttpStatus.FORBIDDEN, nae);
+        }
+        try {
+            permissionChecker.checkReadOrganisation(parentOid);
+        } catch (NotAuthorizedException nae) {
+            LOG.warn("Not authorized to read organisation: " + parentOid);
+            throw new OrganisaatioResourceException(HttpStatus.FORBIDDEN, nae);
+        }
```
Comment (jollem) [inline organisaatio-service/src/main/java/fi/vm/sade/organisaatio/resource/impl/v4/OrganisaatioResourceImplV4.java:234]: repetition

File: organisaatio-service/src/main/java/fi/vm/sade/organisaatio/resource/impl/v4/OrganisaatioResourceImplV4.java [hunk 2]
```diff
@@ -208,6 +217,54 @@ public OrganisaatioHakutulosV4 findDescendants(String oid) {
...
+        Date date;
+        if (dateParam != null && dateParam.getValue() != null) {
+            date = dateParam.getValue();
+        } else {
+            date = new Date();
+        }
```
Comment (jollem) [inline organisaatio-service/src/main/java/fi/vm/sade/organisaatio/resource/impl/v4/OrganisaatioResourceImplV4.java:240]: this _could_ be separate method (probably has some use elsewhere also)

File: organisaatio-ui/src/components/Modaalit/ToimipisteenYhdistys/TYBody.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import React, { useContext } from 'react';
```
Comment (jollem) [inline organisaatio-ui/src/components/Modaalit/ToimipisteenYhdistys/TYBody.tsx:1]: About filename: Is the "TY" prefix needed? These files are in dedicate module after all?

File: organisaatio-ui/src/types/types.ts [hunk 4]
```diff
@@ -191,3 +198,8 @@ export type TranslatedInputBind = {
 export type FrontProperties = {
     urlVirkailija: string;
 };
+
+export interface iOption {
```
Comment (jollem) [inline organisaatio-ui/src/types/types.ts:202]: "i" Propbably stads for interface? is the prefix needed?

File: organisaatio-ui/src/components/Sivut/LomakeSivu/LomakeSivu.tsx [hunk 5]
```diff
@@ -128,6 +160,8 @@ const LomakeSivu = ({ match: { params }, history }: LomakeSivuProps) => {
 
     if (
         !organisaatio ||
+        historiaLoading ||
```
Comment (jollem) [inline organisaatio-ui/src/components/Sivut/LomakeSivu/LomakeSivu.tsx:163]: just a suggestion:
`if ( fetchingData() ) { return <Spinner/> }`
Note: missing localization

File: organisaatio-service/src/main/java/fi/vm/sade/organisaatio/resource/impl/v4/OrganisaatioResourceImplV4.java [hunk 6]
```diff
@@ -208,6 +217,54 @@ public OrganisaatioHakutulosV4 findDescendants(String oid) {
...
+
+        OrganisaatioRDTOV4 organisaatio = this.organisaatioFindBusinessService.findByIdV4(oid, false);
+        OrganisaatioRDTOV4 newParent = this.organisaatioFindBusinessService.findByIdV4(parentOid,false);
+
+        if (organisaatio == null) {
+            throw new OrganisaatioNotFoundException(oid);
+        }
+        if (newParent == null) {
+            throw new OrganisaatioNotFoundException(parentOid);
+        }
```
Comment (jollem) [inline organisaatio-service/src/main/java/fi/vm/sade/organisaatio/resource/impl/v4/OrganisaatioResourceImplV4.java:250]: consider separate method also for this (repetition)

**Review by mkamka — APPROVED** [review]

File: organisaatio-ui/src/components/Modaalit/ToimipisteenYhdistys/TYBody.tsx (hunk 3 above)
Comment (mikaelbarlund) [reply organisaatio-ui/src/components/Modaalit/ToimipisteenYhdistys/TYBody.tsx:1]: This followed convention in the project for other components.

**Review by jollem — APPROVED** [review]

**Pull request merged**